sympy/solvers/recurr.py

```diff
@@ -727,20 +728,19 @@ def rsolve(f, y, init=None):
         coeff = S.One
         kspec = None
         for h in Mul.make_args(g):
-            if h.is_Function:
-                if h.func == y.func:
-                    result = h.args[0].match(n + k)
+            if y.func not in map(lambda x: x.func, preorder_traversal(h)):
```

Comment:
Is this just `if not h.has(y.func):`?

Reply:
Right, I didn't know about such a method. Thank you!